### PR TITLE
UserStory #45 - Edit business info

### PR DESCRIPTION
**Pull request by therobotisnotatoy**

### CODE REVIEW

File: beauty/api/permissions.py [hunk 1]
```diff
@@ -1,49 +1,53 @@
+"""This module provides all needed permissions."""
```
Comment (Misha86) [inline beauty/api/permissions.py:1]: need empty line after module docstring

File: beauty/api/serializers/business_serializers.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+"""The module includes serializers for Business model."""
```
Comment (Misha86) [inline beauty/api/serializers/business_serializers.py:1]: need empty line after module docstring

File: beauty/api/views.py [hunk 3]
```diff
@@ -1,77 +1,84 @@
+"""This module provides all needed views."""
```
Comment (Misha86) [inline beauty/api/views.py:1]: need empty line after module docstring

File: beauty/api/views.py [hunk 4]
```diff
@@ -86,24 +93,78 @@ def perform_destroy(self, instance):
...
+    # permission_classes = [AllowAny]
+
+    queryset = Business.objects.all()
+    serializer_class = BusinessesSerializer
+
+    logger.debug("A view to display businesses of certain owner opened")
+
+    def list(self, request, owner_id): # noqa
+        """Filter businesses of certain owner."""
+        self.permission_classes = (IsAdminOrBusinessOwner,)
```
Comment (Misha86) [inline beauty/api/views.py:144]: did you set permissions in line 134?

**Review by Misha86 — CHANGES_REQUESTED** [review]

File: beauty/api/serializers/business_serializers.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+logger = logging.getLogger(__name__)
+
+
+class BusinessesSerializer(serializers.ModelSerializer):
+    """Serializer for business base fields."""
+
+    name = serializers.CharField(max_length=20)
+    type = serializers.CharField(max_length=100) # noqa
+    address = serializers.CharField(max_length=500)
+
```
Comment (morento101) [inline beauty/api/serializers/business_serializers.py:18]: Is there a need to write name and type fields if Django parses them from the model? You have listed these fields in class Meta.

File: beauty/api/serializers/business_serializers.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+class BusinessAllDetailSerializer(serializers.ModelSerializer):
+    """Serializer for specific business."""
+
+    owner_name = serializers.SerializerMethodField()
+    created_at = serializers.ReadOnlyField()
+    address = serializers.CharField(max_length=500)
+
+    def get_owner_name(self, obj):
+        """Return full name of business owner."""
+        full_name = f"{obj.owner.first_name} {obj.owner.last_name}"
```
Comment (morento101) [inline beauty/api/serializers/business_serializers.py:35]: User model has method get_full_name(). No need to duplicate code

File: beauty/api/serializers/business_serializers.py (hunk 5 above)
Comment (morento101) [reply beauty/api/serializers/business_serializers.py:18]: Does address change?  In DB address is implemented as a chain of tables. In the Business model, the address filed stores FK from the Address table. Can we store Varchar there?

**Review by morento101 — CHANGES_REQUESTED** [review]
Need to change some code

**Pull request closed without merging**